apps/mobile/src/libs/components/text/text.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactNode } from "react";
+import { Text as RNText, type StyleProp, type TextStyle } from "react-native";
+
+import {
+	BaseColor,
+	FontFamilies,
+	FontSize,
+	LineHeight,
+	WeightMap,
+} from "~/libs/enums/enums";
+
+import { styles as presetStyles } from "./styles";
+
+type Properties = {
+	children: ReactNode;
+	color?: (typeof BaseColor)[keyof typeof BaseColor];
+	fontFamily?: keyof typeof FontFamilies;
+	preset?: "default" | "heading" | "subheading";
+	size?: keyof typeof FontSize;
+	style?: StyleProp<TextStyle>;
+	weight?: keyof typeof WeightMap;
+};
+
+const Text = ({
+	children,
+	color = BaseColor.BLACK,
+	preset: presetProperties = "default",
+	size = "md",
+	style: styleOverride,
+	weight = "regular",
+}: Properties) => {
+	const baseStyle = presetStyles[presetProperties];
+
+	const sizeToStyleMap = {
+		fontFamily: "NUNITO",
```

Comment:
```suggestion
		fontFamily: fontWeightToFamilyMap[weight],
```

and remove `fontFamily` from component Properties

Reply:
Done